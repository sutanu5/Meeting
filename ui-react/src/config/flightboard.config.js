module.exports = {
  'board' : {
    'nextUp' : 'Next Up',
    'statusAvailable' : 'Open',
    'statusBusy' : 'Busy',
    'statusError' : 'Error'
  },

  'navbar' : {
    'title' : 'Conference Room Availability',
  },

  'roomFilter' : {
    'filterTitle' : 'Locations',
    'filterAllTitle' : 'All Conference Rooms',
  },
};
