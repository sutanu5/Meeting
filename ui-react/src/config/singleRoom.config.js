module.exports = {
  'nextUp' : 'Next Up',
  'statusAvailable' : 'Open',
  'statusBusy' : 'Busy',
  'upcomingTitle' : 'Upcoming',
};
