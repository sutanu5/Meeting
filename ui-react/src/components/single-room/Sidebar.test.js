import Sidebar from './Sidebar';
import * as config from '../../config/singleRoom.config.js';

describe('Sidebar Component', () => {

  let room, details;
  
  beforeEach(() => {
    room = {
      'Roomlist':'Test Roomlist',
      'Name':'Test Room',
      'RoomAlias':'test-room',
      'Email':'email@email.com',
      'Appointments':[
         {
            'Subject':'Meeting Subject',
            'Organizer':'Meeting Organizer',
            'Start':1532966400000,
            'End':1533344400000
         },
         {
          'Subject':'Meeting Two Subject',
          'Organizer':'Meeting Two Organizer',
          'Start':1532966400000,
          'End':1533344400000
       },
      ],
      'Busy':true
    };

    details = {
      appointmentExists: true,
      upcomingAppointments: false,
      nextUp: ''
    };
  });

  // item.Start
  // item.End

  it('renders correctly', () => {
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders Clock component', () => {
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const component = wrapper.find('Clock').exists();
    expect(component).toBeTruthy();
  });

  it('renders upcomingTitle according to config file', () => {
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const div = wrapper.find('#upcoming-title').text();
    expect(div).toBe(config.upcomingTitle);
  });

  it('renders a table row if details.upcomingAppointments is true', () => {
    details.upcomingAppointments = true;
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const table = wrapper.find('table').text();
    expect(table).not.toBeEmpty();
  });

  it('does not render a table row if details.upcomingAppointments is false', () => {
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const table = wrapper.find('table').text();
    expect(table).toBeEmpty();
  });

  it('renders room.Appointments[1].Subject when details.upcomingAppointments is true' , () => {
    details.upcomingAppointments = true;
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const div = wrapper.find('.up__meeting-title').text();
    expect(div).toBe(room.Appointments[1].Subject);
  });

  it('does not render room.Appointments[1].Subject when details.upcomingAppointments is false' , () => {
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const div = wrapper.find('.up__meeting-title').exists();
    expect(div).toBeFalsy();
  });

  it('renders meeting time when details.upcomingAppointments is true' , () => {
    details.upcomingAppointments = true;
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const div = wrapper.find('.up__meeting-time').text();
    expect(div).not.toBeEmpty();
  });

  it('renders meeting time empty when times are missing' , () => {
    details.upcomingAppointments = true;
    room.Appointments[1].Start = '';
    room.Appointments[1].End = '';
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const div = wrapper.find('.up__meeting-time').text();
    expect(div).toBeEmpty();
  });

  it('does not render meeting time when details.upcomingAppointments is false' , () => {
    const wrapper = shallow(<Sidebar room={room} details={details} config={config} />);

    const div = wrapper.find('.up__meeting-time').exists();
    expect(div).toBeFalsy();
  });
});
