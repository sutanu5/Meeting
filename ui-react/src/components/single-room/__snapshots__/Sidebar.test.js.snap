// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component renders correctly 1`] = `
<div
  className="columns small-4 right-col"
>
  <div
    id="single-room__clock-wrap"
  >
    <Clock />
  </div>
  <div
    id="upcoming-title"
  >
    Upcoming
  </div>
  <table />
</div>
`;
